kernel/kernel-api/src/main/java/io/delta/kernel/Transaction.java

```diff
@@ -59,6 +59,13 @@ public interface Transaction {
    */
   List<String> getPartitionColumns(Engine engine);
 
+  /**
+   * Gets the latest version used as the base of this transaction.
+   *
+   * @return The version of the table as of the beginning of this Transaction
+   */
+  long getCommitBaseVersion();
```

Comment:
Not sure if we want to do a `getCommitBaseVersion` or a `getCommitVersion`. I'm leaning more towards base because retries may make `getCommitVersion` inaccurate.

Thoughts @vkorukanti?

Reply:
or `getReadTableVersion`? we get the version the txn is build on top of. Lets start with this one and if anyone has better names, we can change it (given the change is a small anyways).